ams_app/views.py

```diff
@@ -0,0 +1,444 @@
+import os
+
+import datetime
+from django.conf import settings
+from django.shortcuts import render, redirect
+from django.contrib.auth.models import User
+from django.contrib import messages
+from django.contrib.auth import authenticate, login, logout
+from django.contrib.auth.decorators import login_required
+from django.core.paginator import Paginator, EmptyPage, PageNotAnInteger
+from django.contrib.postgres.search import SearchVector, SearchQuery
+
+from ams_app import forms
+from ams_app.models import Assignment, Submission, Profile
+
+
+def sign_up(request):
+    user_form = forms.UserForm
+    profile_form = forms.ProfileForm
+    if request.method == "POST":
+        user_form = user_form(request.POST)
+        profile_form = profile_form(request.POST)
+        # check whether it's valid:
+        if user_form.is_valid() and profile_form.is_valid():
+            user = User.objects.create_user(
+                username=user_form.cleaned_data["username"],
+                email=user_form.cleaned_data["email"],
+                password=user_form.cleaned_data["password"])
+            user.profile.role = profile_form.cleaned_data["role"]
+            user.profile.matric_number= profile_form.cleaned_data["matric_number"]
+            user.save()
+            messages.success(request, 'User was successfully created.')
+            return redirect('/')
```

Comment:
Would be nice to `login` a user before redirecting. 
Check this https://github.com/laddeos/django-bucketlist-application/blob/master/djangobucketlist/bucketlist/views/view_authentication.py#L66-L69